python/llama-index-server/llama_index/server/tools/index/query.py

```diff
@@ -1,30 +1,58 @@
+import logging
 import os
 from typing import Any, Optional
 
 from llama_index.core.base.base_query_engine import BaseQueryEngine
-from llama_index.core.tools.query_engine import QueryEngineTool
 from llama_index.core.indices.base import BaseIndex
+from llama_index.core.prompts import PromptTemplate
+from llama_index.core.response_synthesizers import Accumulate
+from llama_index.core.tools.query_engine import QueryEngineTool
+from llama_index.server.prompts import CITATION_PROMPT
+from llama_index.server.tools.index.node_citation_processor import NodeCitationProcessor
 
+logger = logging.getLogger(__name__)
 
-def create_query_engine(index: BaseIndex, **kwargs: Any) -> BaseQueryEngine:
+
+def create_query_engine(
+    index: BaseIndex, enable_citation: bool = True, **kwargs: Any
+) -> BaseQueryEngine:
     """
     Create a query engine for the given index.
 
     Args:
         index: The index to create a query engine for.
+        enable_citation: Whether to enable citation in the response of the tool.
         params (optional): Additional parameters for the query engine, e.g: similarity_top_k
     """
     top_k = int(os.getenv("TOP_K", 0))
     if top_k != 0 and kwargs.get("filters") is None:
         kwargs["similarity_top_k"] = top_k
 
+    if enable_citation:
+        if kwargs.get("response_synthesizer") is not None:
+            # We don't override the provided response synthesizer
+            # Just show a warning
+            logger.warning(
+                "Custom response synthesizer and citation are both used. The citation might not work as intended."
```

Comment:
i guess the citation won't work at all - better throw an error?

Reply:
as `Accumulate` is also not the default afaik - how about we log an info that we use Accumulate?